VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room.component.ts

```diff
@@ -0,0 +1,689 @@
+    }
+
+    onJoinConfirmAnswered(actionConfirmed: boolean) {
+        this.logger.debug(`${this.componentLoggerPrefix} Judge responded to join hearing confirmation`, {
+            conference: this.conferenceId,
+            status: this.vhConference.status,
+            confirmStart: actionConfirmed
+        });
+        this.displayJoinHearingPopup = false;
+        if (actionConfirmed) {
+            this.joinHearingInSession();
+        }
+    }
+
+    joinHearingInSession() {
+        this.store.dispatch(VideoCallHostActions.joinHearing({ conferenceId: this.vhConference.id, participantId: this.vhParticipant.id }));
+    }
+
+    audioRestartCallback(continueWithNoRecording: boolean) {
+        this.continueWithNoRecording = continueWithNoRecording;
+        this.audioErrorRetryToast = null;
+    }
+
+    verifyAudioRecordingStream() {
+        /// If audio recording is required,
+        // has not been confirmed by user, to continue without recording,
+        // video is open,
+        // the alert isn't open already,
+        // Recording is not paused
+        // and the audio streaming agent cannot be validated, then show the alert
+        if (
+            this.vhConference.audioRecordingRequired &&
+            !this.continueWithNoRecording &&
+            this.showVideo &&
+            !this.audioErrorRetryToast &&
+            !this.recordingPaused &&
+            !this.audioRecordingService.wowzaAgent?.isAudioOnlyCall
+        ) {
+            this.logWowzaAlert();
+            this.showAudioRecordingRestartAlert();
+        }
+    }
+
+    private onShouldReload(): void {
+        window.location.reload();
+    }
+
+    private onShouldUnload(): void {
+        this.cleanUp();
+    }
+
+    private setUpSubscribers() {
+        this.subscribeToClock();
+        this.startEventHubSubscribers();
+        this.connectToPexip();
+        this.registerMediaStatusPublisher();
+        this.startVideoCallEventSubscribers();
+    }
+
+    private registerMediaStatusPublisher() {
+        this.userMediaService.isAudioOnly$.pipe(takeUntil(this.destroyedSubject)).subscribe(async audioOnly => {
+            this.audioOnly = audioOnly;
+
+            const mediaStatus = new ParticipantMediaStatus(false, audioOnly);
+            await this.eventService.sendMediaStatus(this.conferenceId, this.vhParticipant.id, mediaStatus);
+        });
+    }
+
+    private setTitle() {
+        this.store.dispatch(ConferenceActions.enterWaitingRoom({ userRole: this.userRole }));
+    }
+
+    private cleanUp() {
+        this.logger.debug(`${this.componentLoggerPrefix} clearing intervals and subscriptions`, {
+            conference: this.vhConference?.id
+        });
+
+        this.executeWaitingRoomCleanup();
+
+        this.destroyedSubject.next();
+        this.destroyedSubject.complete();
+        this.stopVideoCallEventSubscribers();
+    }
+
+    private startVideoCallEventSubscribers() {
+        this.subscriptions.push(
```

Comment:
Can we please see the back of subscriptions.push. It's from my early days of not understanding how takeUntil works. That way we're consistent in the file and ideally all components

Reply:
yep, replaced with `.pipe(takeUntil(this.destroyedSubject))`